tools/ics_rst_from_html.py

```diff
@@ -0,0 +1,155 @@
+#
+# auto-pts - The Bluetooth PTS Automation Framework
+#
+# Copyright (c) 2021, Codecoup.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms and conditions of the GNU General Public License,
+# version 2, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+
+"""ICS Launch Studio html to rst file parser
+
+Log into your bluetooth.com account. Go to
+https://launchstudio.bluetooth.com/MyProjects/DraftProjects/
+and select your project. Go to ICS Selection tab and select profile, e.g. GAP.
+Save the page of the profile as html file.
+
+Run this script as follows:
+
+cd auto-pts
+python3 ics_rst_from_html.py GAP path/to/Launch-Studio-ICS-Selection.html
+
+In directory of run will be created gap-pics.rst file.
+"""
+
+import sys
+import os
+import re
+import pandas as pd
+
+
+def make_table(_tab, _profile):
+    grid = [['Parameter Name', 'Selected', 'Description']]
+
+    for i in range(len(_tab.values)):
+        new_row = []
+        parameter_name = 'TSPC_%s_%s' % (_profile, _tab.Item[i].replace('/', '_'))
+        selected = 'True' if _tab.Support[i] else 'False'
+        description = '%s (%s)' % (_tab.Capability[i], _tab.Status[i])
+
+        new_row.append(parameter_name)
+        new_row.append(selected)
+        new_row.append(description)
+        grid.append(new_row)
+
+    max_cols = [max(out) for out in map(list, zip(*[[len(str(item)) for item in row] for row in grid]))]
+
+    rst = table_div(max_cols, 1)
+    rst += normalize_row(grid[0], max_cols)
+
+    rst += table_div(max_cols, 1)
+    for row in grid[1:]:
+        rst += normalize_row(row, max_cols)
+    rst += table_div(max_cols, 1)
+    return rst
+
+
+def table_div(max_cols, header_flag=1):
+    out = ""
+    if header_flag == 1:
+        style = "="
+    else:
+        style = "-"
+
+    for max_col in max_cols:
+        out += max_col * style + " "
+
+    out = out.strip() + "\n"
+    return out
+
+
+def normalize_row(row, max_cols):
+    r = ""
+    for i, max_col in enumerate(max_cols):
+        try:
+            r += row[i] + (max_col - len(row[i]) + 1) * " "
+        except Exception as e:
```

Comment:
Do we need this here? If there is a possibility of an exception we should implement this in such a way as to not trigger it. And definitely not handle generic Exception, at least a specific exception that is triggered.

Reply:
Sorry, it's just debugging leftovers.